k8s/kubicle/kubicle.go

```diff
@@ -0,0 +1,213 @@
+	RAM          int
+	Disk         int
+	User         string
+	ImageUUID    string
+	HTTPSProxy   string
+	HTTPProxy    string
+	NoProxy      string
+	Token        string
+	PublicKey    string
+	UserDataFile string
+	Description  string
+}
+
+type proxyConfig struct {
+	httpProxy  string
+	httpsProxy string
+	noProxy    string
+}
+
+func init() {
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage of %s:\n\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "%s create image-uuid [options]\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "%s delete\n\n", os.Args[0])
+		fmt.Fprintln(os.Stderr, "- create : creates a k8s cluster")
+		fmt.Fprintln(os.Stderr, "- delete : removes kubicle created instances and workloads")
+	}
+}
+
+func createFlags() (*options, error) {
+	opts := options{
+		masterVM: vmOptions{
+			vCPUs: 1,
+			mem:   1024,
+			disk:  10,
+		},
+		workerVM: vmOptions{
+			vCPUs: 1,
+			mem:   2048,
+			disk:  10,
+		},
+		workers: 1,
+	}
+
+	opts.user = os.Getenv("USER")
+	home := os.Getenv("HOME")
+	if home != "" {
+		opts.publicKeyPath = path.Join(home, "local", "testkey.pub")
+	}
+
+	fs := flag.NewFlagSet("create", flag.ExitOnError)
+
+	fs.IntVar(&opts.masterVM.mem, "mmem", opts.masterVM.mem,
+		"Megabytes of RAM allocated to master VM")
+	fs.IntVar(&opts.masterVM.vCPUs, "mcpus", opts.masterVM.vCPUs, "VCPUs assignged to master VM")
+	fs.IntVar(&opts.masterVM.disk, "mdisk", opts.masterVM.disk,
+		"Gigabytes of disk allocated to master VM")
+
+	fs.IntVar(&opts.workerVM.mem, "wmem", opts.workerVM.mem,
+		"Megabytes of RAM allocated to worker VMs")
+	fs.IntVar(&opts.workerVM.vCPUs, "wcpus", opts.workerVM.vCPUs, "VCPUs assignged to worker VM")
+	fs.IntVar(&opts.workerVM.disk, "wdisk", opts.workerVM.disk,
+		"Gigabytes of disk allocated to worker VM")
+
+	fs.IntVar(&opts.workers, "workers", opts.workers, "Number of worker nodes to create")
+
+	fs.StringVar(&opts.publicKeyPath, "key", opts.publicKeyPath, "Path to public key used to ssh into nodes")
+	fs.StringVar(&opts.user, "user", opts.user, "Name of user account to create on the nodes")
+	fs.StringVar(&opts.externalIP, "external-ip", opts.externalIP,
+		"External-ip to associate with the master node")
+	fs.BoolVar(&opts.keep, "keep", false, "Retains workload definition files if set to true")
+
+	if err := fs.Parse(flag.Args()[1:]); err != nil {
+		return nil, err
+	}
+
+	if len(fs.Args()) < 1 {
+		return nil, usageError("No image-uuid specified!")
+	}
+	opts.imageUUID = fs.Args()[0]
+
+	return &opts, nil
+}
+func runCommand(signalCh <-chan os.Signal) error {
+	var err error
+
+	errCh := make(chan error)
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	switch os.Args[1] {
+	case "create":
+		go create(ctx, errCh)
+	case "delete":
+		go destroy(ctx, errCh)
+	}
+	select {
+	case <-signalCh:
+		cancelFunc()
+		err = <-errCh
+	case err = <-errCh:
+		cancelFunc()
+	}
+
+	return err
+}
+
+// TODO: copied from launcher
+func getLock() error {
+	err := os.MkdirAll(lockDir, 0777)
```

Comment:
Agreed.  Also I'm thinking of getting rid of this altogether.  It will prevent a race if you try to create two k8s clusters in the same tenant at the same time from the same machine.  But it can't help if two people run kubicle create on the same tenant from two different machines at the same time.  What do you think?  Is it worth keeping?